tests/pytest/test_unique_clear.py

```diff
@@ -0,0 +1,18 @@
+def test_unique_clears(testdir):
+    """Successive uses of the `faker` pytest fixture have the
+    generated unique values cleared between functions."""
+
+    testdir.makepyfile(
```

Comment:
I think it is better if you use a similar test you used in `TestUniquenessClass`, and I am thinking something like this:


```python
def test_fully_exhaust_unique_booleans(faker, num_samples):
    _dummy = [faker.boolean() for _ in range(num_samples)]
    faker.unique.boolean()
    faker.unique.boolean()
    with pytest.raises(UniquenessException):
        faker.unique.boolean()
    _dummy = [faker.boolean() for _ in range(num_samples)]

def test_do_not_exhaust_booleans(faker):
    faker.unique.boolean()

def test_fully_exhaust_booleans_again(faker, num_samples):
    _dummy = [faker.boolean() for _ in range(num_samples)]
    faker.unique.boolean()
    faker.unique.boolean()
    with pytest.raises(UniquenessException):
        faker.unique.boolean()
    _dummy = [faker.boolean() for _ in range(num_samples)]
```

As far as I know, `pytest` should run the tests in that order, but if the order is inconsistent, then you can probably use `makeconftest` and declare `pytest_collection_modifyitems` within to enforce the proper test execution order.

Reply:
I'm confused though, pytest suggest in their docs to use the [testdir](https://docs.pytest.org/en/stable/reference.html#pytest.pytester.Testdir) fixture to test plugins, so I'm not sure I understand the difference between what I've wrriten, and what you're suggesting.

Reply:
I am not saying not to use the `testdir` fixture. What I am saying is that the tests you made inside `testdir.makepyfile` can be made better, because you only have these tests inside:

```python
def test_generate_two_unique_booleans(faker):
    faker.unique.boolean()
    faker.unique.boolean()
def test_another_boolean(faker):
    faker.unique.boolean()
```

Reply:
Right, I'm following, apologies. :sweat_smile: